docs/dotnet/dotnet-kestrel-metrics.md

```diff
@@ -0,0 +1,474 @@
+# Semantic Conventions for Kestrel web server metrics emitted by .NET
+
+**Status**: [Experimental, Feature-freeze][DocumentStatus]
+
+This document defines semantic conventions for Kestrel web server.
+
+<!-- toc -->
+
+- [Kestrel endpoint](#kestrel-endpoint)
+- [Metric: `kestrel.active_connections`](#metric-kestrelactive_connections)
+- [Metric: `kestrel.connection.duration`](#metric-kestrelconnectionduration)
+- [Metric: `kestrel.rejected_connections`](#metric-kestrelrejected_connections)
+- [Metric: `kestrel.queued_connections`](#metric-kestrelqueued_connections)
+- [Metric: `kestrel.queued_requests`](#metric-kestrelqueued_requests)
+- [Metric: `kestrel.upgraded_connections`](#metric-kestrelupgraded_connections)
+- [Metric: `kestrel.tls_handshake.duration`](#metric-kestreltls_handshakeduration)
+- [Metric: `kestrel.active_tls_handshakes`](#metric-kestrelactive_tls_handshakes)
+
+<!-- tocstop -->
+
+## Kestrel endpoint
+
+Kestrel endpoint is represented with [`System.Net.EndPoint`](https://learn.microsoft.com/en-us/dotnet/api/system.net.endpoint) class,
+which does not always provide information about server address or port.
+
+Instrumentation supports `IPEndPoint`, `UnixDomainSocketEndPoint`, and `NamedPipeEndPoint` and sets `server.address`, `server.port` (for IP endpoint), `network.type`, and `network.transport` attributes from corresponding endpoint on Kestrel metrics.
+
+In case instrumentation does not recognize `EndPoint` implementation, it sets `server.address` attribute to `endpoint.ToString()` value
+and `network.transport` value to corresponding `endpoint.AddressFamily` property.
+
+## Metric: `kestrel.active_connections`
+
+<!-- semconv metric.kestrel.active_connections(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `kestrel.active_connections` | UpDownCounter | `{connection}` | Number of connections that are currently active on the server. [1] |
+
+**[1]:** Meter name: `Microsoft.AspNetCore.Server.Kestrel`; Added in: ASP.NET Core 8.0
+<!-- endsemconv -->
+
+<!-- semconv metric.kestrel.active_connections(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`network.transport`](../general/attributes.md) | string | [OSI transport layer](https://osi-model.com/transport-layer/) or [inter-process communication method](https://en.wikipedia.org/wiki/Inter-process_communication). [1] | `tcp`; `unix` | Recommended |
```

Comment:
```suggestion
| [`network.transport`](../general/attributes.md) | string | [OSI transport layer](https://osi-model.com/transport-layer/) or [inter-process communication method](https://wikipedia.org/wiki/Inter-process_communication). [1] | `tcp`; `unix` | Recommended |
```
This would be a good global search and replace. I assume you copied this from opentelemetry. We only use wikipedia if there are no alternatives.